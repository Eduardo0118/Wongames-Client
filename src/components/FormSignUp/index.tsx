import TextField from 'components/TextField'
import Link from 'next/link'
import { Email, Lock } from 'styled-icons/material-outlined'
import { UserCircle } from '@styled-icons/fa-regular'
import { FormWrapper, FormLink } from 'components/Form'
import Button from 'components/Button'

const FormSignUp = () => (
  <FormWrapper>
    <form>
      <TextField
        name="name"
        placeholder="Name"
        type="text"
        icon={<UserCircle />}
      />
      <TextField
        name="email"
        placeholder="E-mail"
        type="email"
        icon={<Email />}
      />
      <TextField
        name="password"
        placeholder="Password"
        type="password"
        icon={<Lock />}
      />
      <TextField
        name="confirmPassword"
        placeholder="Confirm password"
        type="password"
        icon={<Lock />}
      />

      <Button size="large" fullWidth>
        Sign up now
      </Button>

      <FormLink>
        Already have an account?{' '}
        <Link href="/sign-in">
          <a>Sign in</a>
        </Link>
      </FormLink>
    </form>
  </FormWrapper>
)

export default FormSignUp
