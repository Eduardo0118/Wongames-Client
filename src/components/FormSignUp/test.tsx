import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormSignUp from '.'

describe('<FormSignUp />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<FormSignUp />)

    expect(container.firstChild).toMatchSnapshot()

    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/e-mail/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Confirm password')).toBeInTheDocument()

    expect(
      screen.getByRole('button', { name: /sign up now/i })
    ).toBeInTheDocument()
  })

  it('should render the text to sign in', () => {
    renderWithTheme(<FormSignUp />)

    expect(screen.getByRole('link', { name: /Sign up/i })).toBeInTheDocument()
    expect(screen.getByText(/Already have an account/i)).toBeInTheDocument()
  })
})
