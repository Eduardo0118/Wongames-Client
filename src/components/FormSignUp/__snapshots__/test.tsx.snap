// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the heading 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c5 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 > svg {
  width: 100%;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: 0;
  cursor: pointer;
  color: #FAFAFA;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c7:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c8 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
}

.c8 a {
  color: #3CD3C1;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.1rem solid #3CD3C1;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #29b3a3;
  border-bottom: 0.1rem solid #29b3a3;
}

<div
  class="c0"
>
  <form>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
              fill="currentColor"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="email"
          placeholder="E-mail"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity="0.87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity="0.87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="c6 c7"
    >
      <span>
        Sign up now
      </span>
    </button>
    <div
      class="c8"
    >
      Already have an account?
       
      <a
        href="/sign-in"
      >
        Sign in
      </a>
    </div>
  </form>
</div>
`
